app/src/main/java/com/github/polypoly/app/ui/map/MapUI.kt

```diff
@@ -1,40 +1,69 @@
 package com.github.polypoly.app.ui.map
 
+import android.content.Context
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.livedata.observeAsState
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.viewinterop.AndroidView
 import com.github.polypoly.app.BuildConfig
 import com.github.polypoly.app.base.game.location.LocationPropertyRepository
+import com.github.polypoly.app.base.game.location.PropertyLevel
 import com.github.polypoly.app.models.game.GameViewModel
 import org.osmdroid.config.Configuration
 import org.osmdroid.views.MapView
 
 object MapUI {
-
-    // store the map view for testing purposes
-    lateinit var mapView: MapView private set
-
     /**
      * Initialize the map view that sits beneath the UI components.
      * @param mapViewModel The view model for the map.
      */
     @Composable
     fun MapView(mapViewModel: MapViewModel, gameViewModel: GameViewModel?) {
+        val mapViewState = mapViewModel.mapViewState.value
         AndroidView(
             factory = { context ->
-                Configuration.getInstance().userAgentValue = BuildConfig.APPLICATION_ID
-                val mapView = initMapView(context)
-                for (zone in LocationPropertyRepository.getZones()) {
-                    for (location in zone.locationProperties) {
-                        addMarkerTo(mapView, location, zone.color, mapViewModel, gameViewModel)
-                    }
-                }
-                val currentLocationOverlay = initLocationOverlay(mapView, mapViewModel, gameViewModel)
-                mapView.overlays.add(currentLocationOverlay)
-                this.mapView = mapView
+                if (gameViewModel != null)
+                    mapViewModel.updateMapViewState(gameViewModel.getGameData().value!!.inGameLocations)
+                val mapView = createMapView(context, mapViewModel, gameViewModel, mapViewState)
                 mapView
-            }, modifier = Modifier.testTag("map"))
+            }, modifier = Modifier.testTag("map")
+        )
+    }
+
+    private fun createMapView(
```

Comment:
same, it is more lisible to put one parameter per line

Reply:
done